### PR TITLE
Create jlm::tests namespace

### PR DESCRIPTION
**Pull request by phate**
Introduce jlm::tests namespace for all test helper files.

### CODE REVIEW

File: tests/TestRvsdgs.cpp [hunk 1]
```diff
@@ -331,8 +333,7 @@ Bits2PtrTest::SetupRvsdg()
   auto nf = graph->node_normal_form(typeid(jlm::rvsdg::operation));
   nf->set_mutable(false);
 
-  auto SetupBit2PtrFunction = [&]()
-  {
+  auto SetupBit2PtrFunction = [&]() {
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:336]: Looks like the curly brackets isn't following coding standard

File: tests/TestRvsdgs.cpp [hunk 2]
```diff
@@ -817,10 +818,9 @@ IndirectCallTest1::SetupRvsdg()
   };
 
   auto SetupTestFunction = [&](
-    lambda::output * fctindcall,
-    lambda::output * fctthree,
-    lambda::output * fctfour)
-  {
+    lambda::output *fctindcall,
+    lambda::output *fctthree,
+    lambda::output *fctfour) {
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:823]: Same here

File: tests/TestRvsdgs.cpp [hunk 3]
```diff
@@ -2192,9 +2192,8 @@ PhiTest2::SetupRvsdg()
   };
 
   auto SetupPhi = [&](
-    lambda::output & lambdaEight,
-    lambda::output & lambdaI)
-  {
+    lambda::output &lambdaEight,
+    lambda::output &lambdaI) {
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:2196]: and here.

File: tests/TestRvsdgs.cpp (hunk 1 above)
Comment (sjalander) [inline tests/TestRvsdgs.cpp:336]: Looks like the curly brackets isn't following coding standard

File: tests/TestRvsdgs.cpp [hunk 4]
```diff
@@ -2883,10 +2885,9 @@ MemcpyTest::SetupRvsdg()
   };
 
   auto SetupFunctionG = [&](
-    delta::output & localArray,
-    delta::output & globalArray,
-    lambda::output & lambdaF)
-  {
+    delta::output &localArray,
+    delta::output &globalArray,
+    lambda::output &lambdaF) {
```
Comment (sjalander) [inline tests/TestRvsdgs.cpp:2890]: and here

File: tests/TestRvsdgs.hpp [hunk 5]
```diff
@@ -7,34 +7,34 @@
...
 /**
 * \brief RvsdgTest class
 */
 class RvsdgTest {
 public:
-	jlm::llvm::RvsdgModule &
-	module()
-	{
+  jlm::llvm::RvsdgModule &
+  module() {
```
Comment (sjalander) [inline tests/TestRvsdgs.hpp:18]: Coding standard

File: tests/TestRvsdgs.hpp [hunk 6]
```diff
@@ -618,110 +603,92 @@ class IndirectCallTest1 final : public RvsdgTest {
 class IndirectCallTest2 final : public RvsdgTest {
 public:
   [[nodiscard]] jlm::llvm::delta::node &
-  GetDeltaG1() const noexcept
-  {
+  GetDeltaG1() const noexcept {
```
Comment (sjalander) [inline tests/TestRvsdgs.hpp:606]: These following changes are not following the coding standard.
Or is it different for different cases?

**Review by sjalander — APPROVED** [review]

File: tests/TestRvsdgs.cpp (hunk 2 above)
Comment (phate) [reply tests/TestRvsdgs.cpp]: Overhauled the TestRvsdgs.hpp and TestRvsdgs.cpp

File: tests/TestRvsdgs.cpp (hunk 1 above)
Comment (phate) [reply tests/TestRvsdgs.cpp]: Overhauled the TestRvsdgs.hpp and TestRvsdgs.cpp

File: tests/TestRvsdgs.cpp (hunk 3 above)
Comment (phate) [reply tests/TestRvsdgs.cpp]: Overhauled the TestRvsdgs.hpp and TestRvsdgs.cpp

File: tests/TestRvsdgs.cpp (hunk 4 above)
Comment (phate) [reply tests/TestRvsdgs.cpp]: Overhauled the TestRvsdgs.hpp and TestRvsdgs.cpp

**Pull request merged**